win32/src/threadpool_win32.c

```diff
@@ -221,77 +233,56 @@ THREADPOOL_WORK_ITEM_HANDLE threadpool_create_work_item(THANDLE(THREADPOOL) thre
-        work_item_context = malloc(sizeof(THREADPOOL_WORK_ITEM));
-        if (work_item_context == NULL)
+        /* Codes_SRS_THREADPOOL_WIN32_05_006: [ Otherwise threadpool_create_work_item shall allocate a context threadpool_work_item_context of type THREADPOOL_WORK_ITEM_CONTEXT_HANDLE where work_function, work_function_context, and ptp_work shall be saved. ]*/
+        threadpool_work_item_context_ptr = THANDLE_MALLOC(THREADPOOL_WORK_ITEM_CONTEXT)(threadpool_dispose_work_item);
+        if (threadpool_work_item_context_ptr == NULL)
         {
             /* Codes_SRS_THREADPOOL_WIN32_05_007: [ If any error occurs, threadpool_create_work_item shall fail and return a NULL value. ]*/
             LogError("malloc failed");
         }
         else
         {
-            work_item_context->work_function = work_function;
-            work_item_context->work_function_context = work_function_context;
+            threadpool_work_item_context_ptr->work_function = work_function;
+            threadpool_work_item_context_ptr->work_function_context = work_function_context;
 
-            /* Codes_SRS_THREADPOOL_WIN32_05_008: [ threadpool_create_work_item shall create work_item_context member variable ptp_work of type PTP_WORK by calling CreateThreadpoolWork to set the callback function as on_work_callback_v2. ]*/
-            work_item_context->ptp_work = CreateThreadpoolWork(on_work_callback_v2, work_item_context, &threadpool_ptr->tp_environment);
-            /* Codes_SRS_THREADPOOL_WIN32_05_009: [ If there are no errors then this work_item_context of type THREADPOOL_WORK_ITEM_HANDLE would be returned indicating a succcess to the caller. ]*/
-            if (work_item_context->ptp_work == NULL)
+            /* Codes_SRS_THREADPOOL_WIN32_05_008: [ threadpool_create_work_item shall create threadpool_work_item_context member variable ptp_work of type PTP_WORK by calling CreateThreadpoolWork to set the callback function as on_work_callback_v2. ]*/
+            threadpool_work_item_context_ptr->ptp_work = CreateThreadpoolWork(on_work_callback_v2, threadpool_work_item_context_ptr, &threadpool_ptr->tp_environment);
+            /* Codes_SRS_THREADPOOL_WIN32_05_009: [ If there are no errors then this threadpool_work_item_context of type THREADPOOL_WORK_ITEM_CONTEXT_HANDLE would be returned indicating a succcess to the caller. ]*/
+            if (threadpool_work_item_context_ptr->ptp_work == NULL)
             {
                 /* Codes_SRS_THREADPOOL_WIN32_05_010: [ If any error occurs, threadpool_create_work_item shall fail, free the newly created context and return a NULL value. ]*/
                 LogError("CreateThreadpoolWork failed");
-                free(work_item_context);
-                work_item_context = NULL;
+                THANDLE_FREE(THREADPOOL_WORK_ITEM_CONTEXT)(threadpool_work_item_context_ptr);
+                threadpool_work_item_context_ptr = NULL;
             }

```

Comment:
> } [](http://example.com/codeflow?start=11&length=2)

nit: should have an "else" branch for readability, even if it just has a comment like "ok"

Reply:
Fixed